visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -129,15 +135,28 @@ void MotionPlanningFrame::computePlanButtonClicked()
 void MotionPlanningFrame::computeExecuteButtonClicked()
 {
   if (move_group_ && current_plan_)
-    move_group_->execute(*current_plan_);
+  {
```

Comment:
Do you really need to run `computeExecuteButtonClicked()` and `computeStopButtonClicked()` in a separate thread (as a background job)? As far as I can see, these methods require only short ROS service call interaction.
Directly executing those commands in `executeButtonClicked()` and `stopButtonClicked()` would simplify the code.
However, the ROS service might be not available.